plugins/in_syslog/syslog_prot.c

```diff
@@ -31,88 +33,128 @@ static inline void consume_bytes(char *buf, int bytes, int length)
     memmove(buf, buf + bytes, length - bytes);
 }
 
-static int append_raw_message_to_record_data(char **result_buffer,
-                                             size_t *result_size,
-                                             flb_sds_t raw_message_key_name,
-                                             char *base_object_buffer,
-                                             size_t base_object_size,
-                                             char *raw_message_buffer,
-                                             size_t raw_message_size)
+static int append_message_to_record_data(char **result_buffer,
+                                         size_t *result_size,
+                                         flb_sds_t message_key_name,
+                                         char *base_object_buffer,
+                                         size_t base_object_size,
+                                         char *message_buffer,
+                                         size_t message_size,
+                                         int message_type)
 {
-    int                i;
     int                result;
-    size_t             unpacker_offset;
-    msgpack_sbuffer    mp_sbuf;
-    msgpack_packer     mp_pck;
-    msgpack_unpacked   unpacked_buffer;
+    char              *modified_data_buffer;
+    int                modified_data_size;
+    msgpack_object_kv *new_map_entries[1];
+    msgpack_object_kv  message_entry;
     *result_buffer = NULL;
     *result_size = 0;
+    modified_data_buffer = NULL;
 
-    unpacker_offset = 0;
-    msgpack_unpacked_init(&unpacked_buffer);
-    result = msgpack_unpack_next(&unpacked_buffer,
-                                 base_object_buffer,
-                                 base_object_size,
-                                 &unpacker_offset);
+    if (message_key_name != NULL) {
+        new_map_entries[0] = &message_entry;
 
-    if (result != MSGPACK_UNPACK_SUCCESS) {
-        return -1;
-    }
+        message_entry.key.type = MSGPACK_OBJECT_STR;
+        message_entry.key.via.str.size = flb_sds_len(message_key_name);
+        message_entry.key.via.str.ptr  = message_key_name;
 
-    if (unpacker_offset != base_object_size) {
-        msgpack_unpacked_destroy(&unpacked_buffer);
-        return -2;
-    }
+        if (message_type == MSGPACK_OBJECT_BIN) {
+            message_entry.val.type = MSGPACK_OBJECT_BIN;
+            message_entry.val.via.bin.size = message_size;
+            message_entry.val.via.bin.ptr  = message_buffer;
+        }
+        else if (message_type == MSGPACK_OBJECT_STR) {
+            message_entry.val.type = MSGPACK_OBJECT_BIN;
```

Comment:
This type seems to be wrong

Reply:
Right. Fixed in https://github.com/fluent/fluent-bit/pull/7651/commits/2810b8e6a64f677e29038c2a84a61712fee3df92.